Doesn't work with proxy

It looks like you don't use IDEA's proxy settings so it doesn't work. Also the whole IDE hags in this case because it couldn't connect.

I have the proxy configured in IDE:
![image](https://github.com/MCMicS/jenkins-control-plugin/assets/143802875/6f631c88-d72e-4eb4-a753-86ae1a91aa46)
When I test proxy settings using the button on the page and using jenkins url it succeeds.

When I click Test Connection in the jenkins server settings page I get in idea.log:
`2023-09-01 13:28:07,552 [3466489]   INFO - #org.codinjutsu.tools.jenkins.security.DefaultSecurityClient - IO Error during retrieve last Host for Redirection for URL 'https://server_name': Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
org.apache.http.conn.ConnectTimeoutException: Connect to seliius01120.seli.gic.ericsson.se:443 [server_name/10.120.212.164] failed: Connect timed out
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.executeHttp(DefaultSecurityClient.java:251)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.getLastRedirectionHost(DefaultSecurityClient.java:183)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.doAuthentication(BasicSecurityClient.java:61)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.connect(BasicSecurityClient.java:51)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.testAuthenticate(RequestManager.java:319)
	at org.codinjutsu.tools.jenkins.settings.ServerConfigurable.testConnection(ServerConfigurable.java:71)
	at org.codinjutsu.tools.jenkins.settings.ServerComponent.testConnection(ServerComponent.java:111)
	at org.codinjutsu.tools.jenkins.settings.ServerComponent.lambda$new$0(ServerComponent.java:71)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6657)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3385)
	at java.desktop/java.awt.Component.processEvent(Component.java:6422)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4954)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4581)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4522)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2808)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:794)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:766)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:764)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.kt:633)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:588)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:121)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
	at java.desktop/java.awt.Dialog.lambda$show$2(Dialog.java:1081)
	at java.desktop/sun.awt.SunToolkit.performOnMainThreadIfNeeded(SunToolkit.java:2164)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1041)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:737)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:462)
	at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1685)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1643)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.kt:104)
	at com.intellij.ide.actions.ShowSettingsAction.perform(ShowSettingsAction.java:60)
	at com.intellij.ide.actions.ShowSettingsAction.actionPerformed(ShowSettingsAction.java:47)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.java:339)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAwareWithCallbacks$4(ActionUtil.java:313)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:362)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:313)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.invokeAction(ActionUtil.java:538)
	at com.intellij.ui.popup.ActionPopupStep.performAction(ActionPopupStep.java:266)
	at com.intellij.ui.popup.ActionPopupStep.lambda$onChosen$3(ActionPopupStep.java:243)
	at com.intellij.ui.popup.AbstractPopup.lambda$dispose$18(AbstractPopup.java:1625)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$3(FocusManagerImpl.java:173)
	at com.intellij.util.ui.EdtInvocationManager.invokeLaterIfNeeded(EdtInvocationManager.java:33)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.kt:210)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.kt:176)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:169)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:163)
	at com.intellij.ui.popup.AbstractPopup.dispose(AbstractPopup.java:1623)
	at com.intellij.ui.popup.WizardPopup.dispose(WizardPopup.java:165)
	at com.intellij.ui.popup.list.ListPopupImpl.dispose(ListPopupImpl.java:394)
	at com.intellij.openapi.util.ObjectTree.runWithTrace(ObjectTree.java:129)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:161)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:262)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:250)
	at com.intellij.ui.popup.WizardPopup.disposeAllParents(WizardPopup.java:273)
	at com.intellij.ui.popup.list.ListPopupImpl.disposePopup(ListPopupImpl.java:496)
	at com.intellij.ui.popup.list.ListPopupImpl.handleNextStep(ListPopupImpl.java:523)
	at com.intellij.ui.popup.list.ListPopupImpl._handleSelect(ListPopupImpl.java:483)
	at com.intellij.ui.popup.list.ListPopupImpl.handleSelect(ListPopupImpl.java:429)
	at com.intellij.ui.popup.list.ListPopupImpl$MyMouseListener.mouseReleased(ListPopupImpl.java:699)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6657)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3385)
	at com.intellij.ui.popup.list.ListPopupImpl$MyList.processMouseEvent(ListPopupImpl.java:764)
	at java.desktop/java.awt.Component.processEvent(Component.java:6422)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4954)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4581)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4522)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2808)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:794)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:766)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:764)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.kt:633)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:588)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 152 more
2023-09-01 13:28:17,563 [3476500]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 20012ms, details saved to C:\Users\xxberusm\AppData\Local\JetBrains\IntelliJIdea2023.2\log\threadDumps-freeze-20230901-132802-IU-232.9559.62-BasicSecurityClient.doAuthentication-20sec
2023-09-01 13:28:28,699 [3487636]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-09-01 13:28:38,403 [3497340]   INFO - #org.codinjutsu.tools.jenkins.security.DefaultSecurityClient - IO Error during retrieve last Host for Redirection for URL 'https://server_name': Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
org.apache.http.conn.ConnectTimeoutException: Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.executeHttp(DefaultSecurityClient.java:251)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.getLastRedirectionHost(DefaultSecurityClient.java:183)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.doAuthentication(BasicSecurityClient.java:61)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.connect(BasicSecurityClient.java:51)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.authenticate(RequestManager.java:300)
	at org.codinjutsu.tools.jenkins.logic.LoginService$1.run(LoginService.java:42)
	at org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask.run(JenkinsBackgroundTask.java:41)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:478)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 37 more`

Here is the log with the custom plugin:

`2023-09-07 16:36:36,303 [ 361456]   INFO - #org.codinjutsu.tools.jenkins.security.DefaultSecurityClient - IO Error during retrieve last Host for Redirection for URL 'https://server_name': Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
org.apache.http.conn.ConnectTimeoutException: Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.executeHttp(DefaultSecurityClient.java:260)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.getLastRedirectionHost(DefaultSecurityClient.java:186)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.doAuthentication(BasicSecurityClient.java:61)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.connect(BasicSecurityClient.java:51)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.testAuthenticate(RequestManager.java:319)
	at org.codinjutsu.tools.jenkins.settings.ServerConfigurable.testConnection(ServerConfigurable.java:71)
	at org.codinjutsu.tools.jenkins.settings.ServerComponent.testConnection(ServerComponent.java:129)
	at org.codinjutsu.tools.jenkins.settings.ServerComponent.lambda$new$1(ServerComponent.java:83)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6657)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3385)
	at java.desktop/java.awt.Component.processEvent(Component.java:6422)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4954)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4581)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4522)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2808)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:794)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:766)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:764)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.kt:633)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:588)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:121)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
	at java.desktop/java.awt.Dialog.lambda$show$2(Dialog.java:1081)
	at java.desktop/sun.awt.SunToolkit.performOnMainThreadIfNeeded(SunToolkit.java:2164)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1041)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:737)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:462)
	at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1685)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1643)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.kt:104)
	at com.intellij.ide.actions.ShowSettingsAction.perform(ShowSettingsAction.java:60)
	at com.intellij.ide.actions.ShowSettingsAction.actionPerformed(ShowSettingsAction.java:47)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.doPerformActionOrShowPopup(ActionUtil.java:339)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAwareWithCallbacks$4(ActionUtil.java:313)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:362)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:313)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.invokeAction(ActionUtil.java:538)
	at com.intellij.ui.popup.ActionPopupStep.performAction(ActionPopupStep.java:266)
	at com.intellij.ui.popup.ActionPopupStep.lambda$onChosen$3(ActionPopupStep.java:243)
	at com.intellij.ui.popup.AbstractPopup.lambda$dispose$18(AbstractPopup.java:1625)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$3(FocusManagerImpl.java:173)
	at com.intellij.util.ui.EdtInvocationManager.invokeLaterIfNeeded(EdtInvocationManager.java:33)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.kt:210)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.kt:176)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:169)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:163)
	at com.intellij.ui.popup.AbstractPopup.dispose(AbstractPopup.java:1623)
	at com.intellij.ui.popup.WizardPopup.dispose(WizardPopup.java:165)
	at com.intellij.ui.popup.list.ListPopupImpl.dispose(ListPopupImpl.java:394)
	at com.intellij.openapi.util.ObjectTree.runWithTrace(ObjectTree.java:129)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:161)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:262)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:250)
	at com.intellij.ui.popup.WizardPopup.disposeAllParents(WizardPopup.java:273)
	at com.intellij.ui.popup.list.ListPopupImpl.disposePopup(ListPopupImpl.java:496)
	at com.intellij.ui.popup.list.ListPopupImpl.handleNextStep(ListPopupImpl.java:523)
	at com.intellij.ui.popup.list.ListPopupImpl._handleSelect(ListPopupImpl.java:483)
	at com.intellij.ui.popup.list.ListPopupImpl.handleSelect(ListPopupImpl.java:429)
	at com.intellij.ui.popup.list.ListPopupImpl$MyMouseListener.mouseReleased(ListPopupImpl.java:699)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6657)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3385)
	at com.intellij.ui.popup.list.ListPopupImpl$MyList.processMouseEvent(ListPopupImpl.java:764)
	at java.desktop/java.awt.Component.processEvent(Component.java:6422)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4954)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4581)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4522)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2808)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:794)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:766)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:764)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.kt:633)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:588)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 152 more
2023-09-07 16:36:41,577 [ 366730]   WARN - #c.i.i.s.e.c.EventLogStatisticsService - java.io.IOException: HTTP/1.1 header parser received no bytes
java.io.IOException: HTTP/1.1 header parser received no bytes
	at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:586)
	at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:123)
	at com.intellij.internal.statistic.eventLog.connection.request.StatsRequestBuilder.trySend(StatsRequestBuilder.java:115)
	at com.intellij.internal.statistic.eventLog.connection.request.StatsRequestBuilder.trySend(StatsRequestBuilder.java:110)
	at com.intellij.internal.statistic.eventLog.connection.request.StatsRequestBuilder.send(StatsRequestBuilder.java:95)
	at com.intellij.internal.statistic.eventLog.connection.SettingsConnectionService.loadSettings(SettingsConnectionService.java:74)
	at com.intellij.internal.statistic.eventLog.connection.SettingsConnectionService.lambda$new$0(SettingsConnectionService.java:43)
	at com.intellij.internal.statistic.eventLog.connection.StatisticsCachingSupplier.get(StatisticsCachingSupplier.java:24)
	at com.intellij.internal.statistic.eventLog.connection.SettingsConnectionService.getExternalSettings(SettingsConnectionService.java:68)
	at com.intellij.internal.statistic.eventLog.connection.EventLogUploadSettingsService.isSettingsReachable(EventLogUploadSettingsService.java:65)
	at com.intellij.internal.statistic.eventLog.connection.EventLogStatisticsService.send(EventLogStatisticsService.java:101)
	at com.intellij.internal.statistic.eventLog.connection.EventLogStatisticsService.send(EventLogStatisticsService.java:57)
	at com.intellij.internal.statistic.updater.StatisticsJobsSchedulerKt$runEventLogStatisticsService$2$1.invokeSuspend(StatisticsJobsScheduler.kt:90)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
Caused by: java.io.IOException: HTTP/1.1 header parser received no bytes
	at java.net.http/jdk.internal.net.http.common.Utils.wrapWithExtraDetail(Utils.java:348)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.onReadError(Http1Response.java:675)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver.checkForErrors(Http1AsyncReceiver.java:302)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver.flush(Http1AsyncReceiver.java:268)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$LockingRestartableTask.run(SequentialScheduler.java:205)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(SequentialScheduler.java:149)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(SequentialScheduler.java:230)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.io.EOFException: EOF reached while reading
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver$Http1TubeSubscriber.onComplete(Http1AsyncReceiver.java:596)
	at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$ReadSubscription.signalCompletion(SocketTube.java:640)
	at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$InternalReadSubscription.read(SocketTube.java:845)
	at java.net.http/jdk.internal.net.http.SocketTube$SocketFlowTask.run(SocketTube.java:181)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(SequentialScheduler.java:230)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(SequentialScheduler.java:303)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler.runOrSchedule(SequentialScheduler.java:256)
	at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$InternalReadSubscription.signalReadable(SocketTube.java:774)
	at java.net.http/jdk.internal.net.http.SocketTube$InternalReadPublisher$ReadEvent.signalEvent(SocketTube.java:957)
	at java.net.http/jdk.internal.net.http.SocketTube$SocketFlowEvent.handle(SocketTube.java:253)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.handleEvent(HttpClientImpl.java:979)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.lambda$run$3(HttpClientImpl.java:934)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.run(HttpClientImpl.java:934)
2023-09-07 16:36:46,342 [ 371495]   WARN - #c.i.d.PerformanceWatcherImpl - UI was frozen for 20081ms, details saved to C:\Users\xxberusm\AppData\Local\JetBrains\IntelliJIdea2023.2\log\threadDumps-freeze-20230907-163631-IU-232.9559.62-BasicSecurityClient.doAuthentication-20sec
2023-09-07 16:37:01,395 [ 386548]  FINER - #org.codinjutsu.tools.jenkins.security.DefaultSecurityClient - Sending HEAD request to: https://server_name
2023-09-07 16:37:01,781 [ 386934]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-09-07 16:37:02,402 [ 387555]   INFO - #c.i.c.ComponentStoreImpl - Saving appAbbreviationManager took 83 ms, FileTypeManager took 39 ms
2023-09-07 16:37:02,604 [ 387757]   INFO - #c.i.c.ComponentStoreImpl - Saving Project(name=GitCA, containerState=COMPONENT_CREATED, componentStore=C:\Users\xxberusm\workspace\repos\repository-gitca)RunManager took 50 ms
2023-09-07 16:37:02,796 [ 387949]   INFO - #c.i.o.c.i.s.StoreUtil - saveProjectsAndApp took 1015 ms
2023-09-07 16:37:11,399 [ 396552]   INFO - #org.codinjutsu.tools.jenkins.security.DefaultSecurityClient - IO Error during retrieve last Host for Redirection for URL 'https://server_name': Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
org.apache.http.conn.ConnectTimeoutException: Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.executeHttp(DefaultSecurityClient.java:260)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.getLastRedirectionHost(DefaultSecurityClient.java:186)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.doAuthentication(BasicSecurityClient.java:61)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.connect(BasicSecurityClient.java:51)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.authenticate(RequestManager.java:300)
	at org.codinjutsu.tools.jenkins.logic.LoginService$1.run(LoginService.java:42)
	at org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask.run(JenkinsBackgroundTask.java:41)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:478)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 37 more
2023-09-07 16:37:21,404 [ 406557]   WARN - org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask - org.codinjutsu.tools.jenkins.exception.ConfigurationException: Error during authentication: Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
2023-09-07 16:37:21,405 [ 406558]   FINE - org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask - 
org.codinjutsu.tools.jenkins.exception.ConfigurationException: Error during authentication: Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.doAuthentication(BasicSecurityClient.java:77)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.connect(BasicSecurityClient.java:51)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.authenticate(RequestManager.java:300)
	at org.codinjutsu.tools.jenkins.logic.LoginService$1.run(LoginService.java:42)
	at org.codinjutsu.tools.jenkins.logic.JenkinsBackgroundTask.run(JenkinsBackgroundTask.java:41)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:478)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.http.conn.ConnectTimeoutException: Connect to server_name:443 [server_name/10.120.212.164] failed: Connect timed out
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.executeHttp(DefaultSecurityClient.java:260)
	at org.codinjutsu.tools.jenkins.security.DefaultSecurityClient.executeHttp(DefaultSecurityClient.java:256)
	at org.codinjutsu.tools.jenkins.security.BasicSecurityClient.doAuthentication(BasicSecurityClient.java:67)
	... 26 more
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:551)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:602)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 37 more
`

--- a/gradle.properties
+++ b/gradle.properties
@@ -29,6 +29,7 @@ platformVersionToVerify=232.8660.142
 
 # Plugin Dependencies -> https://plugins.jetbrains.com/docs/intellij/plugin-dependencies.html
 # Example: platformPlugins = com.intellij.java, com.jetbrains.php:203.4449.22, com.intellij.kubernetes
+#platformPlugins=com.intellij.java,com.jetbrains.restClient
 platformPlugins=com.intellij.java
 
 # Java language level used to compile sources and to generate the files for - Java 11 is required since 2020.3

--- a/src/main/java/org/codinjutsu/tools/jenkins/security/DefaultSecurityClient.java
+++ b/src/main/java/org/codinjutsu/tools/jenkins/security/DefaultSecurityClient.java
@@ -86,6 +86,7 @@ class DefaultSecurityClient implements SecurityClient {
                 .setConnectionRequestTimeout(connectionTimout)
                 .setSocketTimeout(connectionTimout);
         this.credentialsProvider = new BasicCredentialsProvider();
+        final var dnsResolver = new JenkinsDnsResolver();
 
         final RequestConfig defaultRequestConfig = requestConfig
                 .setMaxRedirects(10)
@@ -94,14 +95,19 @@ class DefaultSecurityClient implements SecurityClient {
                 .setSSLContext(sslContext)
                 .setDefaultSocketConfig(socketConfig)
                 .setDefaultRequestConfig(defaultRequestConfig)
+                .setDnsResolver(dnsResolver)
                 .setDefaultCredentialsProvider(credentialsProvider)
                 .setRedirectStrategy(new JenkinsRedirectStrategy());
         this.httpClient = httpClientBuilder.build();
         if (useProxySettings) {
             this.configCreator = url -> {
                 final var configForUrl = RequestConfig.copy(defaultRequestConfig);
-                IdeHttpClientHelpers.ApacheHttpClient4.setProxyForUrlIfEnabled(configForUrl, url);
-                IdeHttpClientHelpers.ApacheHttpClient4.setProxyCredentialsForUrlIfEnabled(credentialsProvider, url);
+                final var useSocks = JenkinsConnectionSocketFactory.INSTANCE.prepareContext(url, sslContext,
+                        getHttpClientContext(), dnsResolver);
+                if (!useSocks) {
+                    IdeHttpClientHelpers.ApacheHttpClient4.setProxyForUrlIfEnabled(configForUrl, url);
+                    IdeHttpClientHelpers.ApacheHttpClient4.setProxyCredentialsForUrlIfEnabled(credentialsProvider, url);
+                }
                 return configForUrl.build();
             };
         } else {
@@ -180,7 +186,7 @@ class DefaultSecurityClient implements SecurityClient {
 
     protected final HttpHost getLastRedirectionHost(HttpHost host) {
         final var httpHead = new HttpHead(host.toURI());
-        LOG.trace(String.format("Sending HEAD request to: %s", host.toURI()));
+        LOG.trace(String.format("  %s", host.toURI()));
         try {
             final var context = getHttpClientContext();
             final var response = executeHttp(httpHead);
@@ -249,14 +255,14 @@ class DefaultSecurityClient implements SecurityClient {
     }
 
     protected final HttpResponse executeHttp(HttpPost post) throws IOException {
-        final var postConfig = post.getConfig();
-        if (postConfig == null) {
-            post.setConfig(configCreator.apply(post.getURI().toASCIIString()));
-        }
         return executeHttp((HttpRequestBase) post);
     }
 
     protected final HttpResponse executeHttp(HttpRequestBase httpRequest) throws IOException {
+        final var httpRequestConfig = httpRequest.getConfig();
+        if (httpRequestConfig == null) {
+            httpRequest.setConfig(configCreator.apply(httpRequest.getURI().toASCIIString()));
+        }
         return getHttpClient().execute(httpRequest, this.httpClientContext);
     }
 

--- a/src/main/java/org/codinjutsu/tools/jenkins/security/DefaultSecurityClient.java
+++ b/src/main/java/org/codinjutsu/tools/jenkins/security/DefaultSecurityClient.java
@@ -17,6 +17,7 @@
 package org.codinjutsu.tools.jenkins.security;
 
 import com.intellij.notification.NotificationType;
+import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.util.net.IdeHttpClientHelpers;
 import com.intellij.util.net.ssl.CertificateManager;
@@ -102,8 +103,9 @@ class DefaultSecurityClient implements SecurityClient {
         if (useProxySettings) {
             this.configCreator = url -> {
                 final var configForUrl = RequestConfig.copy(defaultRequestConfig);
-                final var useSocks = JenkinsConnectionSocketFactory.INSTANCE.prepareContext(url, sslContext,
-                        getHttpClientContext(), dnsResolver);
+                final var useSocks = ApplicationManager.getApplication()
+                        .getService(JenkinsConnectionSocketFactory.class)
+                        .prepareContext(url, sslContext, getHttpClientContext(), dnsResolver);
                 if (!useSocks) {
                     IdeHttpClientHelpers.ApacheHttpClient4.setProxyForUrlIfEnabled(configForUrl, url);
                     IdeHttpClientHelpers.ApacheHttpClient4.setProxyCredentialsForUrlIfEnabled(credentialsProvider, url);

--- a/src/main/java/org/codinjutsu/tools/jenkins/security/DefaultSecurityClient.java
+++ b/src/main/java/org/codinjutsu/tools/jenkins/security/DefaultSecurityClient.java
@@ -21,6 +21,7 @@ import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.util.net.IdeHttpClientHelpers;
 import com.intellij.util.net.ssl.CertificateManager;
 import org.apache.commons.lang.StringUtils;
+import org.apache.http.Header;
 import org.apache.http.HttpHost;
 import org.apache.http.HttpResponse;
 import org.apache.http.auth.AuthScope;
@@ -127,6 +128,7 @@ class DefaultSecurityClient implements SecurityClient {
         runMethod(urlStr, data, responseCollector);
 
         if (isRedirection(responseCollector.statusCode)) {
+            LOG.trace(String.format("Handle redirect to: %s", responseCollector.data));
             runMethod(responseCollector.data, data, responseCollector);
         }
 
@@ -178,10 +180,12 @@ class DefaultSecurityClient implements SecurityClient {
 
     protected final HttpHost getLastRedirectionHost(HttpHost host) {
         final var httpHead = new HttpHead(host.toURI());
+        LOG.trace(String.format("Sending HEAD request to: %s", host.toURI()));
         try {
             final var context = getHttpClientContext();
             final var response = executeHttp(httpHead);
             final var statusCode = response.getStatusLine().getStatusCode();
+            LOG.trace(String.format("HEAD response for [%s] with status [%s]", host.toURI(), statusCode));
 
             var targetHost = host;
             if (HttpURLConnection.HTTP_OK == statusCode) {
@@ -211,11 +215,14 @@ class DefaultSecurityClient implements SecurityClient {
         final var post = createPost(url, data);
 
         try {
+            LOG.trace(String.format("Executing POST to: %s", post.getURI()));
             final var response = executeHttp(post);
             final var statusCode = response.getStatusLine().getStatusCode();
             final var responseBody = EntityUtils.toString(response.getEntity());
             final var errorHeader = response.getFirstHeader("X-Error");
             checkResponse(statusCode, responseBody);
+            LOG.trace(String.format("POST response for [%s] with status [%s]: %s", post.getURI(), statusCode,
+                    responseBody));
 
             if (HttpURLConnection.HTTP_OK == statusCode) {
                 responseCollector.collect(statusCode, responseBody);
@@ -227,7 +234,9 @@ class DefaultSecurityClient implements SecurityClient {
             }
             // if redirect handling not works or we still get a redirect header we handle it manually
             if (isRedirection(statusCode)) {
-                responseCollector.collect(statusCode, response.getLastHeader("Location").getValue());
+                final Header location = response.getLastHeader("Location");
+                LOG.trace(String.format("Handle redirect manually to: %s", location));
+                responseCollector.collect(statusCode, location.getValue());
             }
         } catch (UnknownHostException uhEx) {
             throw new ConfigurationException(String.format("Unknown server: %s", uhEx.getMessage()), uhEx);

--- a/src/main/java/org/codinjutsu/tools/jenkins/security/JenkinsConnectionSocketFactory.java
+++ b/src/main/java/org/codinjutsu/tools/jenkins/security/JenkinsConnectionSocketFactory.java
@@ -1,5 +1,6 @@
 package org.codinjutsu.tools.jenkins.security;
 
+import com.intellij.openapi.components.Service;
 import com.intellij.openapi.vfs.VfsUtil;
 import com.intellij.util.net.HttpConfigurable;
 import com.intellij.util.net.IdeaWideProxySelector;
@@ -26,14 +27,13 @@ import java.net.URI;
 import java.util.Collections;
 import java.util.List;
 
-public class JenkinsConnectionSocketFactory {
-    @NotNull
-    public static final JenkinsConnectionSocketFactory INSTANCE = new JenkinsConnectionSocketFactory();
+@Service
+public final class JenkinsConnectionSocketFactory {
 
-    public @Nullable Registry<ConnectionSocketFactory> getRegistry(@NotNull String url, @NotNull SSLContext sslContext) {
+    public @NotNull Registry<ConnectionSocketFactory> getRegistry(@NotNull String url, @NotNull SSLContext sslContext) {
         final Proxy proxy = getProxy(url);
         if(proxy.type() == Proxy.Type.SOCKS) {
-            return createSocksRegistry(toUri(url), proxy, sslContext);
+            return createSocksRegistry(proxy, sslContext);
         }
         return createDefaultRegistry(sslContext);
     }
@@ -71,12 +71,11 @@ public class JenkinsConnectionSocketFactory {
         return false;
     }
 
-    private Registry<ConnectionSocketFactory> createSocksRegistry(@Nullable URI uri,
-                                                                            @NotNull Proxy proxy,
-                                                                            @NotNull SSLContext sslContext) {
+    private Registry<ConnectionSocketFactory> createSocksRegistry(@NotNull Proxy proxy,
+                                                                  @NotNull SSLContext sslContext) {
         return RegistryBuilder.<ConnectionSocketFactory>create()
                 .register("http", new SocksPlainConnectionSocketFactory(proxy))
-                .register("https", new SocksSSLConnectionSocketFactory(uri != null ? uri.getHost() : null, sslContext, proxy))
+                .register("https", new SocksSSLConnectionSocketFactory(sslContext, proxy))
                 .build();
     }
 
@@ -110,16 +109,13 @@ public class JenkinsConnectionSocketFactory {
             this.proxy = proxy;
         }
 
-        @NotNull
-        public Proxy getProxy() {
-            return this.proxy;
-        }
-
+        @Override
         @NotNull
         public Socket createSocket(@Nullable HttpContext context) {
             return new Socket(this.proxy);
         }
 
+        @Override
         @NotNull
         public Socket connectSocket(int connectTimeout,
                                     @Nullable Socket socket,
@@ -133,27 +129,21 @@ public class JenkinsConnectionSocketFactory {
     }
 
     private static final class SocksSSLConnectionSocketFactory extends SSLConnectionSocketFactory {
-        @Nullable
-        private final String host;
         @NotNull
         private final Proxy proxy;
 
-        public SocksSSLConnectionSocketFactory(@Nullable String host, @NotNull SSLContext sslContext, @NotNull Proxy proxy) {
+        public SocksSSLConnectionSocketFactory(@NotNull SSLContext sslContext, @NotNull Proxy proxy) {
             super(sslContext);
-            this.host = host;
             this.proxy = proxy;
         }
 
-        @NotNull
-        public Proxy getProxy() {
-            return this.proxy;
-        }
-
+        @Override
         @NotNull
         public Socket createSocket(@NotNull HttpContext context) {
             return new Socket(this.proxy);
         }
 
+        @Override
         @NotNull
         public Socket connectSocket(int connectTimeout,
                                     @Nullable Socket socket,

--- a/src/main/java/org/codinjutsu/tools/jenkins/security/JenkinsConnectionSocketFactory.java
+++ b/src/main/java/org/codinjutsu/tools/jenkins/security/JenkinsConnectionSocketFactory.java
@@ -1,0 +1,168 @@
+package org.codinjutsu.tools.jenkins.security;
+
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.util.net.HttpConfigurable;
+import com.intellij.util.net.IdeaWideProxySelector;
+import org.apache.http.HttpHost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.socket.PlainConnectionSocketFactory;
+import org.apache.http.conn.ssl.DefaultHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.util.PublicSuffixMatcherLoader;
+import org.apache.http.protocol.HttpContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.Socket;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+
+public class JenkinsConnectionSocketFactory {
+    @NotNull
+    public static final JenkinsConnectionSocketFactory INSTANCE = new JenkinsConnectionSocketFactory();
+
+    public @Nullable Registry<ConnectionSocketFactory> getRegistry(@NotNull String url, @NotNull SSLContext sslContext) {
+        final Proxy proxy = getProxy(url);
+        if(proxy.type() == Proxy.Type.SOCKS) {
+            return createSocksRegistry(toUri(url), proxy, sslContext);
+        }
+        return createDefaultRegistry(sslContext);
+    }
+
+    @NotNull
+    private static Registry<ConnectionSocketFactory> createDefaultRegistry(@NotNull SSLContext sslContext) {
+        String[] supportedProtocols = null;
+        String[] supportedCipherSuites = null;
+        HostnameVerifier proxyAuthStrategyCopy = new DefaultHostnameVerifier(PublicSuffixMatcherLoader.getDefault());
+        final SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(sslContext,
+                supportedProtocols, supportedCipherSuites, proxyAuthStrategyCopy);
+
+        return RegistryBuilder.<ConnectionSocketFactory>create()
+                .register("http", PlainConnectionSocketFactory.getSocketFactory())
+                .register("https", sslConnectionSocketFactory)
+                .build();
+    }
+
+    public boolean prepareContext(@NotNull String url,
+                               @NotNull SSLContext sslContext,
+                               @NotNull HttpClientContext httpClientContext,
+                               @NotNull JenkinsDnsResolver dnsResolver) {
+        httpClientContext.removeAttribute("http.socket-factory-registry");
+        dnsResolver.clearFakes();
+        final URI uri = toUri(url);
+        final Proxy proxy = getProxy(uri);
+        if(proxy.type() == Proxy.Type.SOCKS) {
+            // see org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.SOCKET_FACTORY_REGISTRY
+            httpClientContext.setAttribute("http.socket-factory-registry", getRegistry(url, sslContext));
+            if (uri != null) {
+                dnsResolver.addFakeHost(uri.getHost());
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private Registry<ConnectionSocketFactory> createSocksRegistry(@Nullable URI uri,
+                                                                            @NotNull Proxy proxy,
+                                                                            @NotNull SSLContext sslContext) {
+        return RegistryBuilder.<ConnectionSocketFactory>create()
+                .register("http", new SocksPlainConnectionSocketFactory(proxy))
+                .register("https", new SocksSSLConnectionSocketFactory(uri != null ? uri.getHost() : null, sslContext, proxy))
+                .build();
+    }
+
+    @NotNull
+    private static HttpConfigurable getHttpConfigurable() {
+        return HttpConfigurable.getInstance();
+    }
+
+    private static @NotNull Proxy getProxy(String url) {
+        return getProxy(toUri(url));
+    }
+
+    private static @NotNull Proxy getProxy(@Nullable URI uri) {
+        final List<Proxy> proxies = uri == null ? Collections.emptyList() :
+                new IdeaWideProxySelector(getHttpConfigurable()).select(uri);
+        return proxies.isEmpty() ? Proxy.NO_PROXY : proxies.get(0);
+    }
+
+    @Nullable
+    private static URI toUri(String url) {
+        return url != null ? VfsUtil.toUri(url) : null;
+    }
+
+
+    private static final class SocksPlainConnectionSocketFactory extends PlainConnectionSocketFactory {
+        @NotNull
+        private final Proxy proxy;
+
+        public SocksPlainConnectionSocketFactory(@NotNull Proxy proxy) {
+            super();
+            this.proxy = proxy;
+        }
+
+        @NotNull
+        public Proxy getProxy() {
+            return this.proxy;
+        }
+
+        @NotNull
+        public Socket createSocket(@Nullable HttpContext context) {
+            return new Socket(this.proxy);
+        }
+
+        @NotNull
+        public Socket connectSocket(int connectTimeout,
+                                    @Nullable Socket socket,
+                                    @NotNull HttpHost host,
+                                    @NotNull InetSocketAddress remoteAddress,
+                                    @Nullable InetSocketAddress localAddress,
+                                    @Nullable HttpContext context) throws IOException {
+            InetSocketAddress unresolvedRemote = InetSocketAddress.createUnresolved(host.getHostName(), remoteAddress.getPort());
+            return super.connectSocket(connectTimeout, socket, host, unresolvedRemote, localAddress, context);
+        }
+    }
+
+    private static final class SocksSSLConnectionSocketFactory extends SSLConnectionSocketFactory {
+        @Nullable
+        private final String host;
+        @NotNull
+        private final Proxy proxy;
+
+        public SocksSSLConnectionSocketFactory(@Nullable String host, @NotNull SSLContext sslContext, @NotNull Proxy proxy) {
+            super(sslContext);
+            this.host = host;
+            this.proxy = proxy;
+        }
+
+        @NotNull
+        public Proxy getProxy() {
+            return this.proxy;
+        }
+
+        @NotNull
+        public Socket createSocket(@NotNull HttpContext context) {
+            return new Socket(this.proxy);
+        }
+
+        @NotNull
+        public Socket connectSocket(int connectTimeout,
+                                    @Nullable Socket socket,
+                                    @NotNull HttpHost host,
+                                    @NotNull InetSocketAddress remoteAddress,
+                                    @Nullable InetSocketAddress localAddress,
+                                    @Nullable HttpContext context) throws IOException {
+            final var unresolvedRemote = InetSocketAddress.createUnresolved(host.getHostName(), remoteAddress.getPort());
+            return super.connectSocket(connectTimeout, socket, host, unresolvedRemote, localAddress, context);
+        }
+    }
+}

--- a/src/main/java/org/codinjutsu/tools/jenkins/security/JenkinsDnsResolver.java
+++ b/src/main/java/org/codinjutsu/tools/jenkins/security/JenkinsDnsResolver.java
@@ -1,0 +1,33 @@
+package org.codinjutsu.tools.jenkins.security;
+
+import org.apache.http.conn.DnsResolver;
+import org.apache.http.impl.conn.SystemDefaultDnsResolver;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class JenkinsDnsResolver implements DnsResolver {
+    private static final DnsResolver DNS_RESOLVER = SystemDefaultDnsResolver.INSTANCE;
+    private final Set<String> hostWithFakeDns = new HashSet<>();
+
+    public void clearFakes() {
+        hostWithFakeDns.clear();
+    }
+
+    public void addFakeHost(String host) {
+        hostWithFakeDns.add(host);
+    }
+
+    @Override
+    public InetAddress[] resolve(@NotNull String host) throws UnknownHostException {
+        return hostWithFakeDns.contains(host) ? returnFakeDns(): DNS_RESOLVER.resolve(host);
+    }
+
+    private InetAddress[] returnFakeDns() throws UnknownHostException {
+        final var fakeIp = new byte[]{1, 1, 1, 1};
+        return new InetAddress[]{InetAddress.getByAddress(fakeIp)};
+    }
+}

